cmake/BLTInstallableMacros.cmake

```
@@ -428,37 +428,157 @@ endmacro(blt_add_target_compile_flags)
-            target_include_directories(${arg_TARGET} SYSTEM INTERFACE ${_include_dirs})
+    if(NOT DEFINED arg_TLIST OR NOT DEFINED ${arg_TLIST})
+        message(FATAL_ERROR "TLIST is a required parameter for the blt_find_target_dependencies macro")
+    endif()
+
+    set(_depends_on "")
+
+    # Get dependencies if BLT registered library
+    string(TOUPPER ${arg_TARGET} _target_upper)
+    if(_BLT_${_target_upper}_IS_REGISTERED_LIBRARY)
+        list(APPEND _depends_on "${_BLT_${_target_upper}_DEPENDS_ON}")
+    endif()
+    unset(_target_upper)
+
+    # Get dependencies if CMake target
+    if(TARGET ${arg_TARGET})
+        get_property(_target_type TARGET ${arg_TARGET} PROPERTY TYPE)
+        if(NOT "${_target_type}" STREQUAL "INTERFACE_LIBRARY")
+            get_target_property(_propval ${arg_TARGET} LINK_LIBRARIES)
+            if(_propval)
+                list(APPEND _depends_on ${_propval})
+            endif()
+        endif()
+
+        # get interface link libraries
+        get_target_property(_propval ${arg_TARGET} INTERFACE_LINK_LIBRARIES)
+        if (_propval)
+            list(APPEND _depends_on ${_propval})
         endif()
     endif()
 
-    unset(_include_dirs)
-endmacro()
+    blt_list_remove_duplicates(TO _depends_on)
+    foreach(t ${_depends_on})
+        if (NOT "${t}" IN_LIST ${arg_TLIST})
+            list(APPEND ${arg_TLIST} ${t})
+            blt_find_target_dependencies(TARGET ${t} TLIST ${arg_TLIST})
+        endif()
+    endforeach()
+
+    unset(_depends_on)
+endmacro(blt_find_target_dependencies)
+
+
+##------------------------------------------------------------------------------
+## blt_convert_to_system_includes(TARGETS [<target>...]
+##                                CHILDREN [TRUE|FALSE]
+##                                [QUIET])
+##
+## Converts existing interface includes to system interface includes.
+##------------------------------------------------------------------------------
+function(blt_convert_to_system_includes)
+    set(options QUIET)
+    set(singleValuedArgs CHILDREN TARGET)
+    set(multiValuedArgs TARGETS)
+
+    ## parse the arguments to the macro
+    cmake_parse_arguments(arg
+        "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
+
+    if(DEFINED arg_TARGET)
+        message(WARNING "TARGET is a deprecated parameter for the blt_convert_to_system_includes macro.")
+        list(APPEND arg_TARGETS ${arg_TARGET})
+    endif()
+
+    if(NOT DEFINED arg_TARGETS)
+        message(FATAL_ERROR "TARGETS is a required parameter for the blt_convert_to_system_includes macro.")
+    endif()
+
+    if(NOT DEFINED arg_CHILDREN)
+        set(arg_CHILDREN TRUE)
+    endif()
+
+    # PGI does not support -isystem
+    if(NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "PGI")
+        set(_target_list "")
+
+        foreach(_target ${arg_TARGETS})
+            string(TOUPPER ${_target} _target_upper)
+
+            if(TARGET ${_target} OR ${_BLT_${_target_upper}_IS_REGISTERED_LIBRARY})
```

```suggestion
            if(TARGET ${_target} OR _BLT_${_target_upper}_IS_REGISTERED_LIBRARY)
```

This is going to evaluate to an empty string which causes a parsing error in CMake. Pretty sure this fix is right.

---

I'll trust you on this.